### ISSUE TITLE
CI builds failing after being converted to feedstock

### ISSUE THREAD

**Issue by chambbj** [conda-forge/pdal-feedstock#1]
@conda-forge/core 

The `pdal-feedstock` was created after the conda-smithy issues/upgrade occurred as discussed [here](https://github.com/conda-forge/staged-recipes/issues/5660#issuecomment-382498866).

While the recipe passed all tests back in `conda-forge/staged-recipes`, it has yet to pass any tests after being converted to a feedstock. Linux and macOS both seem to crash following install -- something to do with insufficient permissions with `pdal-config`. It's not obvious to me what's going on here, is it something wrong with the recipe that wasn't handled in the original CI builds? Or something related to the upgrade.

Appreciate your help!

**Comment by dougalsutherland** [comment]
```
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 420, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 411, in execute
    verify=args.verify)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/api.py", line 199, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 2105, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 1515, in build
    built_package = bundlers[output_d.get('type', 'conda')](output_d, m, env, stats)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 917, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/build.py", line 780, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/post.py", line 637, in post_build
    osx_is_app=osx_is_app)
  File "/opt/conda/lib/python3.6/site-packages/conda_build/post.py", line 54, in fix_shebang
    with io.open(path, encoding=locale.getpreferredencoding(), mode='r+') as fi:
PermissionError: [Errno 13] Permission denied: '/home/conda/feedstock_root/build_artifacts/pdal_1524080533998/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/bin/pdal-config'
```

No idea what's causing this....

**Comment by dougalsutherland** [comment]
Oops, still using 3.8.0. I don't know why that would be; `conda-forge-ci-setup` requires `conda-build 3.x`, and so I'd think that `conda install --yes --quiet conda-forge-ci-setup=1` would update to `3.8.1` but it seems like it's staying with the `3.8.0` that's apparently in the Docker image.

Grasping at straws: @conda-forge-admin, please rerender?

**Comment by jakirkham** [comment]
It probably thinks that `conda-build=3` is already satisfied. Triggered a rebuild of the image to get a newer version of `conda-build` included.

**Comment by jakirkham** [comment]
The Docker image is rebuilt.

**Comment by jakirkham** [comment]
My naive guess is `conda-build` thinks `pdal-config` is a text file, which may not be true. `conda-build` has several options along these lines worth exploring.

ref: https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#relocation

cc @msarahan (as improper file detection might be interesting, if that's the issue)

**Comment by dougalsutherland** [comment]
It is a text file: https://github.com/PDAL/PDAL/blob/master/apps/pdal-config.in

**Comment by jakirkham** [comment]
Is it incorrectly identifying a shebang?

**Comment by jakirkham** [comment]
TBH I saw this issue while trying to readd some improperly converted recipes to `staged-recipes` yesterday, but thought we should wait and see what happens since it was presumably passing before and could be some `conda-build` 2/3 issue. It's starting to look like this wasn't building on `conda-build` 2 (so passing might not have meant anything). The `conda-build` 3 issue would suspect is a bug.

**Comment by jakirkham** [comment]
@chambbj, could you please raise this to the `conda-build` issue tracker with links to the logs? Thanks.

**Comment by chambbj** [comment]
Will do. Thanks everybody for taking a look!

**Comment by chambbj** [comment]
@jakirkham This doesn't seem to have received much attention over in the `conda-build` issue tracker (or maybe I'm just being impatient). Any suggestions as to how I can nudge it along? Maybe a workaround we can put in place on the feedstock end in the meantime?

**Comment by chambbj** [comment]
Closed via #3
